### DISCUSSION TITLE
Hi. what the type of a lucide-react icon is?

### DISCUSSION THREAD

**Discussion by ghost** [lucide-icons/lucide#1869] [Help] [answered]
Does anyone know what is the type of a lucide-react icon?

**Comment by zoharma** [comment]
I'm not sure if this is what you meant, but it generates an inline SVG element.

You can find more info here - https://lucide.dev/guide/packages/lucide-react

**Comment by sovetski** [comment, marked as answer]
I think @skdishansachin means in typescript, if we have an object with "icon" key, what type of return we have to add? I also need this

Edit:

Ok I found the solution:

```tsx
import { LucideIcon } from "lucide-react";

export type TestInterface = {
  title: string;
  icon: LucideIcon;
};
```

And when you use your `TestInterface`, you have to use like this:

```tsx
import { ArrowBigUp } from "lucide-react";

const test: TestInterface = {
  title: "Test title",
  icon: ArrowBigUp // be careful, don't use as a component like <ArrowBigUp />
}
```

**Reply by ThalyssonLeite** [reply]
Why not to use? You are being informative with this, not instructive.

**Reply by tolgamorf** [reply]
Yes, this is correct. The correct type is: `React.ForwardRefExoticComponent<Omit<LucideProps, "ref"> & React.RefAttributes<SVGSVGElement>>`, and `LucideIcon` is an alias for this type.

**Reply by Robots-Pixels** [reply]
Thank you very much @sovetski, I ran into the same issue.

@tolgamorf, I actually discovered the same thing by hovering over the imported icon in VS Code, and saw exactly what you mentioned. I’ll make sure to rely on that trick next time. What’s a bit amusing is that we could technically use the full type 
`React.ForwardRefExoticComponent<Omit<LucideProps, "ref"> & React.RefAttributes<SVGSVGElement>>` 

<img width="1010" height="292" alt="Screenshot from 2025-08-23 23-49-48" src="https://github.com/user-attachments/assets/333a676b-93e2-4eec-883d-b50e1631161e" />
  instead of the alias

**Comment by Godcoder696** [comment]
Well you can use `React.JSX.Element` .

Exmaple:
`interface Props{
    icon: React.JSX.Element;
}`

**Comment by Neil-Lunavat** [comment]
Can you share how it works after importing? Do we just go test.icon and it becomes that or do we have to wrap it in <> brackets?

**Reply by ghost** [reply]
What do you mean exactly?

**Comment by Jahid0178** [comment]
If anyone use DynamicIcon component. You can also declear type like this: keyof typeof dynamicIconImports
